Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -1055,7 +1055,7 @@ static inline IntRect computeImageDataRect(ImageBuffer& buffer, int width, int h
 
 void CanvasRenderingContext2DBase::postProcessPixelBuffer() const
 {
-    if (m_wasLastDrawPutImageData || !canvasBase().scriptExecutionContext()->settingsValues().canvasNoiseInjectionEnabled)
+    if (m_wasLastDrawPutImageData || !canvasBase().shouldInjectNoiseBeforeReadback())
```

Comment:
I wonder if we should swap these, if it'll be disabled more often than not - but probably won't make a large perf difference either way.

Reply:
Swapping these is fine with me!